frame/support/src/storage/child.rs

```diff
@@ -19,14 +19,31 @@
 //! This module is a currently only a variant of unhashed with additional `storage_key`.
 //! Note that `storage_key` must be unique and strong (strong in the sense of being long enough to
 //! avoid collision from a resistant hash function (which unique implies)).
+//!
+//! A **key collision free** unique id is required as parameter to avoid key collision
+//! between child tries.
+//! This unique id management and generation responsability is delegated to pallet module.
+//! In the future this id could be generated and stored out of chain state, and using None as a
+//! parameter will result into this mode of operation.
```

Comment:
I think we can remove these lines as it is quite hypotetical. I know I said maybe this could handled by system or something. But at the end I also feel very fine to have modules managing it themself. Like they currently manage their prefix for storages.

Maybe just remove all these optional unique id we will open an issue for ideas.

Later we can make it optional very easily anyway